File: hw4/a.py
def get_synonym(synonyms, word):
    synonyms_dict = {}
    for synonym_word1, synonym_word2 in synonyms:
        synonyms_dict[synonym_word1] = synonym_word2
        synonyms_dict[synonym_word2] = synonym_word1
    return synonyms_dict[word]


assert get_synonym([('Hello', 'Hi'), ('Bye', 'Goodbye'), ('List', 'Array')], 'Goodbye') == 'Bye'
assert get_synonym([('beep', 'Car')], 'Car') == 'beep'
assert get_synonym([('Ololo', 'Ololo'), ('Numbers', '1234567890')], 'Numbers') == '1234567890'


def main():
    n = int(input())
    synonyms = [tuple(input().split()) for _ in range(n)]
    word = input()
    print(get_synonym(synonyms, word))


if __name__ == '__main__':
    main()

File: hw4/a.js
const readline = require('readline');
const assert = require('assert')

const getSynonym = (synonyms, word) => {
    const synonymsDict = new Map()
    for (const pair of synonyms) {
        synonymsDict.set(pair[0], pair[1])
        synonymsDict.set(pair[1], pair[0])
    }

    return synonymsDict.get(word)
}


assert.strictEqual(getSynonym([['Hello', 'Hi'], ['Bye', 'Goodbye'], ['List', 'Array']], 'Goodbye'), 'Bye')
assert.strictEqual(getSynonym([['beep', 'Car']], 'Car'), 'beep')
assert.strictEqual(getSynonym([['Ololo', 'Ololo'], ['Numbers', '1234567890']], 'Numbers'), '1234567890')


const rl = readline.createInterface({ input: process.stdin, output: process.stdout });

const inputLines = [];

rl.on('line', (data) => {
  inputLines.push(data.toString().trim());
});

rl.on('close', () => {
  process.stdout.write(inputProcessing(inputLines));
});


function inputProcessing(lines) {
    const n = lines[0]
    const synonyms = lines.slice(1, lines.length - 1).map(line => line.split(' '))
    const word = lines[lines.length - 1]
    return getSynonym(synonyms, word)
}
